### PR TITLE
Updates for moby 27.0

### PR DESCRIPTION
**Pull request by robmry**
## Description

A collection of updates for the moby 27.0 release ...

### ip6tables enabled by default

`--ip6tables` is now the default, and it no longer needs `--experimental`.

Also, daemon.json `"ipv6":true` only enables IPv6 for the default bridge, it's not required for a user-defined bridge.

- https://github.com/moby/moby/pull/47747

### Update the description of iptables rules

- describe Docker's custom chains
- describe the new direct routing options
- use documentation address ranges in examples

- https://github.com/moby/moby/pull/47871

### kernel module 'ip6_tables' is needed for DinD

- https://github.com/moby/moby/issues/47895

### Explain bridge option host_binding_ipv4

Because it's IPv6 too, despite the option name, and it has surprising differences between v4/v6 handling. Related to ...
- https://github.com/moby/moby/pull/47975

### Updates to IPv6 address pools

A ULA prefix is used if there's no IPv6 in default-address-pools
- https://github.com/moby/moby/pull/47853

Restrictions on IPv6 subnet size and number of pools have been removed
- https://github.com/moby/moby/pull/47768 

## Reviews

@akerouanton, @dvdksn

### CODE REVIEW

File: content/network/drivers/bridge.md [hunk 1]
```diff
@@ -245,9 +241,20 @@ Restart Docker for the changes to take effect.
...
 
-If you configure Docker for IPv6 support (see [Use IPv6](#use-ipv6)), the
-default bridge network is also configured for IPv6 automatically. Unlike
-user-defined bridges, you can't selectively disable IPv6 on the default bridge.
+IPv6 can be enabled for the default bridge using options in `daemon.json`.
+
+```json
+{
+  "ipv6": true,
+  "bip": "192.168.1.1/24",
```
Comment (akerouanton) [inline content/network/drivers/bridge.md:249]: I think we shouldn't add unrelated to parameters to this example (ie. `bip`, `fixed-cidr`, `mtu`, `default-gateway`, `dns`).

File: content/config/daemon/ipv6.md [hunk 2]
```diff
@@ -6,61 +6,26 @@ aliases:
...
-4. Create a new IPv6 network.
-
-   - Using `docker network create`:
+- Using `docker network create`:
 
-     ```console
-     $ docker network create --ipv6 --subnet 2001:0DB8::/112 ip6net
-     ```
+  ```console
+  $ docker network create --ipv6 --subnet 2001:db8::/112 ip6net
```
Comment (akerouanton) [inline content/config/daemon/ipv6.md:16]:
```suggestion
  $ docker network create --ipv6 --subnet 2001:db8::/64 ip6net
```

File: content/config/daemon/ipv6.md [hunk 3]
```diff
@@ -6,61 +6,26 @@ aliases:
...
-            config:
-              - subnet: 2001:0DB8::/112
-     ```
+  ```yaml
+   networks:
+     ip6net:
+       enable_ipv6: true
+       ipam:
+         config:
+           - subnet: 2001:db8::/112
```
Comment (akerouanton) [inline content/config/daemon/ipv6.md:27]:
```suggestion
           - subnet: 2001:db8::/64
```

File: content/network/drivers/bridge.md (hunk 1 above)
Comment (robmry) [reply content/network/drivers/bridge.md]: It was an extension of the example above (which previously mixed v6 params with the other stuff) ... but it probably is clearer without - removed.

File: content/config/daemon/ipv6.md (hunk 2 above)
Comment (robmry) [reply content/config/daemon/ipv6.md]: Done.

File: content/config/daemon/ipv6.md (hunk 3 above)
Comment (robmry) [reply content/config/daemon/ipv6.md]: Done.

**Review by dvdksn — APPROVED** [review]

**Review by akerouanton — APPROVED** [review]

**Review by thaJeztah — DISMISSED** [review]
This is part of the 27.0 release, correct? In that case, we can't merge this yet.

File: content/config/daemon/ipv6.md [hunk 4]
```diff
@@ -214,6 +176,17 @@ subnet supports 65 536 IPv6 addresses.
...
 [wikipedia-ipv6-ula]: https://en.wikipedia.org/wiki/Unique_local_address
 
+## Docker in Docker
+
+On a host using `xtables` (legacy `iptables`) instead of `nftables`, kernel
+module `ip6_tables` must be loaded before an IPv6 Docker network can be created, 
+It is normally loaded automatically when Docker starts.
+
+However, if you running Docker in Docker that is not based on a recent
+[Docker Official Image](https://hub.docker.com/_/docker), you may need to
```
Comment (dvdksn) [inline content/config/daemon/ipv6.md:186]: Maybe reword this to avoid conflating the `docker` image with Docker Official Images

```suggestion
However, if you running Docker in Docker that is not based on a recent
version of the [official `docker` image](https://hub.docker.com/_/docker), you may need to
```

File: content/network/_index.md [hunk 5]
```diff
@@ -132,6 +133,96 @@ it isn't necessary to publish the container's ports.
 You can enable inter-container communication by connecting the containers to the
 same network, usually a [bridge network](./drivers/bridge.md).
 
+If no host IP is given in a port mapping, the bridge network is IPv4-only,
+and `--userland-proxy=true`; ports on the Docker host's IPv6 addresses will
+be mapped to the container's IPv4 address.
```
Comment (dvdksn) [inline content/network/_index.md:138]: Might just be me, but I have a hard time parsing this sentence. Is it saying that if all of these three conditions are true, then IPv6 addresses on the host map to IPv4 addresses of containers? What about something like this, where it leads with the main clause / outcome:

```suggestion
Ports on the host's IPv6 addresses will map to the container's IPv4 address
if no host IP is given in a port mapping, the bridge network is IPv4-only,
and `--userland-proxy=true` (default).
```

File: content/config/daemon/ipv6.md (hunk 4 above)
Comment (robmry) [reply content/config/daemon/ipv6.md]: Done.

File: content/network/_index.md (hunk 5 above)
Comment (robmry) [reply content/network/_index.md]: Much better, thank you.

**Comment by robmry** [comment]
In the last push - I moved the new description of direct routing to the page describing `iptables`, and updated the `iptables` description.

File: content/network/drivers/bridge.md [hunk 6]
```diff
@@ -133,6 +134,27 @@ your own `docker0` bridge. Use this option if you want to run multiple daemon
...
 [Run multiple daemons](../../reference/cli/dockerd.md#run-multiple-daemons).
 
+### Default host binding address
+
+When no host address is given in port publishing options like `-p 80`
+or `-p 8080:80`, the default is to make the container's port 80 available on all
+host addresses, IPv4 and IPv6.
+
+Option `com.docker.network.bridge.host_binding_ipv4` can be used to modify the
+default address for published ports.
```
Comment (dvdksn) [inline content/network/drivers/bridge.md:144]: suggestion: clarify (again) that this is a driver option, since we refer to
a CLI option in the previous paragraph.

```suggestion
The bridge network driver option `com.docker.network.bridge.host_binding_ipv4`
can be used to modify the default address for published ports.
```

File: content/network/drivers/bridge.md [hunk 7]
```diff
@@ -244,9 +262,15 @@ Restart Docker for the changes to take effect.
 
 ### Use IPv6 with the default bridge network
 
-If you configure Docker for IPv6 support (see [Use IPv6](#use-ipv6)), the
-default bridge network is also configured for IPv6 automatically. Unlike
-user-defined bridges, you can't selectively disable IPv6 on the default bridge.
+IPv6 can be enabled for the default bridge using options in `daemon.json`.
```
Comment (dvdksn) [inline content/network/drivers/bridge.md:265]: question: would it makes sense to add a short description for the options we are setting here? I.e. that you need to specify the subnet and default gateway, this example uses example values, you need to change it to something that works for you...

Maybe redundant but as a noob I am not sure what this does, and why all of these are necessary

```suggestion
To enable IPv6 on the default bridge network, configure the following
options in the `daemon.json` configuration file:
```

File: content/network/packet-filtering-firewalls.md [hunk 8]
```diff
@@ -49,84 +65,193 @@ For example:
...
+        "HostIp": "0.0.0.0",
+        "HostPort": "8080"
+      },
+      {
+        "HostIp": "::",
+        "HostPort": ""
+      }
+    ]
+  },
+```
```
Comment (dvdksn) [inline content/network/packet-filtering-firewalls.md:199]:
````suggestion

```console
$ docker container inspect <id> --format "{{json .NetworkSettings.Ports}}"
{"80/tcp":[{"HostIp":"0.0.0.0","HostPort":"8080"},{"HostIp":"::","HostPort":""}]}
```

````

File: content/network/packet-filtering-firewalls.md [hunk 9]
```diff
@@ -49,84 +65,193 @@ For example:
...
 
 > **Important**
 >
 > Using the `conntrack` extension may result in degraded performance.
 { .important }
 
-### Restrict connections to the Docker host
+## Port publishing and mapping
+
+By default, for both IPv4 and IPv6, blocks access to ports that have not
```
Comment (dvdksn) [inline content/network/packet-filtering-firewalls.md:78]:
```suggestion
By default, for both IPv4 and IPv6, the daemon blocks access to ports that have not
```

File: content/network/drivers/bridge.md (hunk 6 above)
Comment (robmry) [reply content/network/drivers/bridge.md]: Done.

File: content/network/drivers/bridge.md (hunk 7 above)
Comment (robmry) [reply content/network/drivers/bridge.md]: Good idea - I've added some words.

File: content/network/packet-filtering-firewalls.md (hunk 8 above)
Comment (robmry) [reply content/network/packet-filtering-firewalls.md]: Done.

File: content/network/packet-filtering-firewalls.md (hunk 9 above)
Comment (robmry) [reply content/network/packet-filtering-firewalls.md]: Done.

**Review by dvdksn — APPROVED** [review]
